reporters/dogstatsd.go

```diff
@@ -84,3 +111,21 @@ func NewDogStatsD(config *viper.Viper, logger *logrus.Logger) (*DogStatsD, error
 func NewDogStatsDFromClient(c dogstatsd.Client, r string) *DogStatsD {
 	return &DogStatsD{client: c, region: r}
 }
+
+func (d *DogStatsD) restartTicker() {
+	for range d.ticker.C {
+		d.mutex.Lock()
+		if err := d.statsdClient.Close(); err != nil {
+			d.logger.Errorf("DogStatsD: failed to close statsd connection during restart: %s", err.Error())
```

Comment:
If you failed to close, shouldn't you return here? Otherwise, you'll keep unclosed resources, potentially leaking memory.

Reply:
You're totally right!

Question: since this is a go routine, we're using a ticker to loop over an interval, should I use `continue` instead?

Reply:
Yes!